### PR TITLE
data model: implement edition statement transformation

### PR DESCRIPTION
**Pull request by reropag**
* implements edition statement transformation

Co-Authored-by: Gianni Pante <gianni.pante@rero.ch>

## Why are you opening this PR?

- implement task: thttps://tree.taiga.io/project/rero21-reroils/task/1144

## How to test?

- the data are not recreated yet, just for code review

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py [hunk 1]
```diff
@@ -173,6 +174,28 @@ def unimarc_to_author(self, key, value):
...
+@unimarctojson.over('editionStatement', '^205..')
+@utils.for_each_value
+@utils.ignore_value
+def unimarc_to_edition_statement(self, key, value):
+    """Get edition statement data.
+
+    editionDesignation: 205 [$a non repetitive] (without trailing ponctuation)
+    responsibility: 205 [$f non repetitive]
+    """
+    key_per_code = {
```
Comment (rerowep) [inline rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py]: Maybe better to understand like this:
```
        subfields_a = utils.force_list(value.get('a'))
        if subfields_a:
            subfield_a = subfields_a[0]
            edition_data['editionDesignation'] = subfield_a
        subfields_f = utils.force_list(value.get('f'))
        if subfields_f:
            subfield_f = subfields_f[0]
            edition_data['responsibility'] = subfield_f
```

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 2]
```diff
@@ -237,6 +270,49 @@ def marc21_to_copyright_date(self, key, value):
...
+        return data
+
+    tag_link, link = get_field_link_data(value)
+    items = get_field_items(value)
+    index = 1
+    edition_data = {}
+    subfield_selection = {'a', 'b'}
+    for blob_key, blob_value in items:
+        if blob_key in subfield_selection:
+            subfield_selection.remove(blob_key)
```
Comment (rerowep) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py]: Could be written also with a `list` and `pop`

File: rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 3]
```diff
@@ -237,6 +270,49 @@ def marc21_to_copyright_date(self, key, value):
...
+
+    editionDesignation: 250 [$a non repetitive] (without trailing /)
+    responsibility: 250 [$b non repetitive]
+    """
+    key_per_code = {
+        'a': 'editionDesignation',
+        'b': 'responsibility'
+    }
+
+    def build_edition_data(code, label, index, link):
```
Comment (rerowep) [inline rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py]: Maybe we have to write a general function usable for other metadata.

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 4]
```diff
@@ -1591,6 +1591,103 @@
         "minLength": 4
```
Comment (rerowep) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: This files get bigger and bigger, maybe delete it and put it in .gitignore. The `setup` script should recreate it if the date is older as source file.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json [hunk 5]
```diff
@@ -1589,6 +1589,103 @@
         "minLength": 4
```
Comment (rerowep) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json]: This files get bigger and bigger, maybe delete it and put it in .gitignore. The setup script should recreate it if the date is older as source file.

File: rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py (hunk 1 above)
Comment (reropag) [reply rero_ils/modules/documents/dojson/contrib/unimarctojson/model.py]: ok

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json (hunk 5 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1.json]: ok

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 6]
```diff
@@ -1591,6 +1591,103 @@
...
       }
     },
+    "editionStatement": {
+      "title": "Edition Statement",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: can you ask the PO for title and description for all the new fields?

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 7]
```diff
@@ -1591,6 +1591,103 @@
...
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "additionalProperties": false,
+        "properties": {
+          "editionDesignation": {
+            "title": "Edition Designation",
+            "type": "array",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: you can put minItems here

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 8]
```diff
@@ -1591,6 +1591,103 @@
...
+                    "und-kore",
+                    "und-zyyy"
+                  ]
+                }
+              }
+            }
+          },
+          "responsibility": {
+            "title": "Responsibility",
+            "type": "array",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: minItems?

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 9]
```diff
@@ -1105,6 +1105,53 @@
...
+      "title": "Edition Statement",
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "additionalProperties": false,
+        "properties": {
+          "editionDesignation": {
+            "title": "Edition Designation",
+            "type": "array",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json]: minItems

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json [hunk 10]
```diff
@@ -1105,6 +1105,53 @@
...
+                  "title": "Language",
+                  "type": "string",
+                  "$ref": "#/definitions/language_script"
+                }
+              }
+            }
+          },
+          "responsibility": {
+            "title": "Responsibility",
+            "type": "array",
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json]: minItems

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json (hunk 8 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: ok, added minItems=1

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json (hunk 6 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: Ok with the PO: "Edition statement"

Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: Ok with PO: Edition statement

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json (hunk 10 above)
Comment (reropag) [reply rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1_src.json]: no min array, responsibility is not mandatory

File: rero_ils/modules/documents/mappings/v6/documents/document.json [hunk 11]
```diff
@@ -189,15 +189,41 @@
...
+                "value": {
+                  "type": "text",
+                  "analyzer": "global_lowercase_asciifolding"
+                },
+                "language": {
+                  "type": "keyword"
+                }
+              }
+            },
+            "responsibility": {
```
Comment (jma) [inline rero_ils/modules/documents/mappings/v6/documents/document.json]: spell

File: rero_ils/modules/documents/mappings/v6/documents/document.json [hunk 12]
```diff
@@ -189,15 +189,41 @@
...
+            }
+          }
+        },
         "provisionActivity": {
           "type": "object",
           "properties": {
             "statement": {
               "type": "object",
               "properties": {
-                "_text": {
```
Comment (jma) [inline rero_ils/modules/documents/mappings/v6/documents/document.json]: why did you remove this?

File: rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json [hunk 13]
```diff
@@ -1591,6 +1591,104 @@
         "minLength": 4
       }
     },
+    "editionStatement": {
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-minimal-v0.0.1.json]: you should remove this file

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 14]
```diff
@@ -1103,6 +1103,54 @@
...
+                },
+                "language": {
+                  "title": "Language",
+                  "type": "string",
+                  "$ref": "#/definitions/language_script"
+                }
+              }
+            }
+          },
+          "responsibility": {
```
Comment (jma) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: spell

File: rero_ils/modules/documents/mappings/v6/documents/document.json (hunk 11 above)
Comment (reropag) [reply rero_ils/modules/documents/mappings/v6/documents/document.json]: it is correct: responsibility and not responsability
